### DISCUSSION TITLE
请教 <n-message> container 的高度设置的原因

### DISCUSSION THREAD

**Discussion by zjlovezj** [tusen-ai/naive-ui#2832] [Q&A] [answered]
在源码中，
https://github.com/TuSimple/naive-ui/blob/cbc2940bd84ea74782b900d2efb2d5cbadedbe1a/src/message/src/styles/index.cssr.ts#L113

<img width="290" alt="image" src="https://user-images.githubusercontent.com/388222/164451057-5b5557c9-7ce1-4a64-983b-89853e1e7506.png">

这行代码在 Chrome/58.0.3029.110 的环境中导致多个message会重叠在一起。

按说 div.n-message-container 的 height 默认就是 auto，这应该能满足需求了，在 Chrome 58 和 Chrome 100 中也表现正常。

所以为什么要设置 height: 0 ?

**Comment by 07akioni** [comment, marked as answer]
> 在源码中，
> 
> https://github.com/TuSimple/naive-ui/blob/cbc2940bd84ea74782b900d2efb2d5cbadedbe1a/src/message/src/styles/index.cssr.ts#L113
> 
> <img alt="image" width="290" src="https://user-images.githubusercontent.com/388222/164451057-5b5557c9-7ce1-4a64-983b-89853e1e7506.png">
> 
> 这行代码在 Chrome/58.0.3029.110 的环境中导致多个message会重叠在一起。
> 
> 按说 div.n-message-container 的 height 默认就是 auto，这应该能满足需求了，在 Chrome 58 和 Chrome 100 中也表现正常。
> 
> 所以为什么要设置 height: 0 ?

因为如果是 auto 的话会因为太宽盖住其他元素。

Chrome/58.0.3029.110 是明确不会支持的版本，不到 70 的肯定会有各种各样的问题。

**Reply by zjlovezj** [reply]
> 因为如果是 auto 的话会因为太宽盖住其他元素。
> 
> Chrome/58.0.3029.110 是明确不会支持的版本，不到 70 的肯定会有各种各样的问题。

谢谢解答。

可惜我现在做的产品大部分还不能升级Chrome内核，因为显卡驱动的兼容性问题和其他兼容性问题。

之前没看到对Chrome版本范围的要求，要是有的话，肯定不敢用了 :)

但是现在产品已经进入测试阶段了，好在没有深度使用naive-ui，更没有定制化，暂时还没发现大的问题。

具体回到本问题，目前我是通过加point-events属性来解决的，因为产品的原因，具体的代码就不贴了。
大概就是给 .n-message-container 设置 point-events: none; 让它即使遮罩其他元素了，也不会影响其他元素的点击。
同时给  .n-message-container * 的后代还原 point-events: auto;

**Reply by 07akioni** [reply]
版本范围一般都不会写的，因为很少有团队会有资源专门找特定的浏览器版本去测试，另外试了确实也不会有啥用，因为也很难有机会把每个 feature 都试一遍。这个东西就和食品的保质期差不多，如果超过了吃也不一定会出问题，但是总有一个推荐的范围。你去翻翻别的库很可能会告诉你 latest 2 version（其实在当今的语境下也没啥意义了，因为版本号翻的太快）。

然后 pointer events 应该是可以的，我感觉可以改一下。
